background.ts

```diff
@@ -43,6 +45,8 @@ let imageEventHandler: ImageEventHandler|null = null;
 let currentContainerEngine = settings.ContainerEngine.NONE;
 let currentImageProcessor: ImageProcessor | null = null;
 let enabledK8s: boolean;
+let pathManager: PathManager = new ManualPathManager();
+const integrationManager: IntegrationManager = getIntegrationManager();
```

Comment:
I suppose you *could* combine them into one big object. But, slicing it this way makes sense to me for the following reasons:

- Smaller objects are more easily understood.
- There is a clean separation between managing the `PATH` variable and managing integrations, both conceptually and in terms of implementation. Each one manages different ideas and different files.
- On macOS and Linux we have two (possibly three in the future) ways of managing the `PATH` variable, and yet only one way of managing symlinks. I suppose we could pass a `PathManager` to an `IntegrationManager` and then call `.enforce()` and `.remove()` on only the `IntegrationManager`, but this would obscure what is going on to a reader of `background.ts`. It would also make it harder to test `IntegrationManager`. It would make the object hierarchy more tree-like and less flat, which seems worse to me.

There may be some confusion because of how I've defined the `PathManager` and `IntegrationManager` interfaces - they are very similar. Perhaps this is a code smell. Perhaps we don't need two separate interfaces, but rather an `IdempotentEnforceRemove` interface, i.e. one that lets you make generic changes on the system and remove them. Or maybe we need to rename their methods to be more specific to `PATH` management or integration management.

Reply:
This reminds me of the [single responsibility principle](https://en.wikipedia.org/wiki/Single-responsibility_principle) and the [interface segregation principle](https://en.wikipedia.org/wiki/Interface_segregation_principle). I hope I'm not being dogmatic though.

Reply:
I'm totally happy with separate managers, just need to have their responsibilities clearly delineated in a way that I understand (and I think the issue here is mostly me).  Here, mostly because `integration` was a superset of `path`, and now that they're distinct I wish to know what else is left in `integration`.

To rephrase — I feel like the responsibility of the integration manager is a bit vague, especially given that path manager _also_ exists. Again, this might just be me being dense…

Having the same methods on the interfaces is fine — that's why we have types to ensure we are clear which is which :)

Reply:
I agree that the idea of an `IntegrationManager` is vague. I can also see how one would think that `PATH` management would fall under its scope, considering the comment on `IntegrationManager`.

I think the vagueness results from my not wanting to impose much of a structure on the Windows side. I haven't spent a lot of time with Windows and feel that I could have missed something that would ruin the whole abstraction. Perhaps if we could be more certain about how Windows is going to work after the refactor, we could figure out a name and description for `IntegrationManager` that are more specific. Right now we could easily call `IntegrationManager` `SymlinkManager` if we just look at how it works on macOS and Linux.

I dunno, maybe I'll have better ideas tomorrow. Maybe we should postpone this discussion until we're working on the Windows version of `IntegrationManager`, or whatever we decide to call it at that time.